### PR TITLE
[21.05] ceph: clean up mon/mgr/rgw dependencies

### PR DESCRIPTION
**Pull request by ctheune**
Required turning the mon and mgr units into simple units.

Fixes PL-132473

@flyingcircusio/release-managers

## Release process

Impact:

n/a

Changelog:

n/a

### PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [x] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [x] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)

Limit unnecessary downtimes.

- [x] Security requirements tested? (EVIDENCE)

Manually tested on cartman06, relying on existing test suite.

### CODE REVIEW

File: pkgs/fc/ceph/src/fc/ceph/mon/nautilus.py [hunk 1]
```diff
@@ -72,26 +70,39 @@ def create(
             run.ceph_authtool(
                 # fmt: off
                 "-g",
-                "-n", "mon.",
-                "--create-keyring", f"{tmpdir}/keyring",
-                "--cap", "mon", "allow *",
+                "-n",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/mon/nautilus.py]: formatting is broken

File: pkgs/fc/ceph/src/fc/ceph/mon/nautilus.py [hunk 2]
```diff
@@ -72,26 +70,39 @@ def create(
...
             )
             # adjust admin capabilities
             run.ceph_authtool(
                 # fmt: off
                 f"{tmpdir}/keyring",
-                "--cap", "mds", "allow *",
-                "--cap", "mon", "allow *",
-                "--cap", "osd", "allow *",
-                "--cap", "mgr", "allow *",
+                "--cap",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/mon/nautilus.py]: same

File: pkgs/fc/ceph/src/fc/ceph/mon/nautilus.py [hunk 3]
```diff
@@ -118,10 +133,13 @@ def create(
         # Create mon on disk structures
         run.ceph_mon(
             # fmt: off
-            "-i", self.id,
+            "-i",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/mon/nautilus.py]: same

File: pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py [hunk 4]
```diff
@@ -331,9 +330,12 @@ def get_largest_free() -> str:
         """Find suitable WAL VG: the one with the most free bytes"""
         return run.json.vgs(
             # fmt: off
-            "-S", "vg_name=~^vgjnl[0-9][0-9]$",
-            "-o", "vg_name,vg_free",
-            "-O", "-vg_free",
+            "-S",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py]: formatting

File: pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py [hunk 5]
```diff
@@ -511,10 +513,15 @@ def _set_crush_and_auth(
         """
         run.ceph(
             # fmt: off
-            "auth", "add", self.name,
-            "osd", "allow *",
-            "mon", "allow rwx",
-            "-i", f"{self.data_volume.mountpoint}/keyring",
+            "auth",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py]: formatting

File: pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py [hunk 6]
```diff
@@ -543,9 +550,10 @@ def run_safety_check(ids: List[int], strict_safety_check: bool):
             except CalledProcessError as e:
                 print(
                     # fmt: off
-                    "OSD not safe to destroy:", e.stderr,
+                    "OSD not safe to destroy:",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py]: formatting

File: pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py [hunk 7]
```diff
@@ -556,9 +564,10 @@ def run_safety_check(ids: List[int], strict_safety_check: bool):
             except CalledProcessError as e:
                 print(
                     # fmt: off
-                    "OSD not okay to stop:", e.stderr,
+                    "OSD not okay to stop:",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py]: formatting

File: pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py [hunk 8]
```diff
@@ -705,11 +714,16 @@ def create(
 
         run.ceph_osd(
             # fmt: off
-            "-i", str(self.id),
-            "--mkfs", "--mkkey",
-            "--osd-objectstore", "bluestore",
-            "--osd-data", self.data_volume.mountpoint,
-            "--bluestore-block-wal-path", self.wal_volume.device,
+            "-i",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py]: formatting

File: pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py [hunk 9]
```diff
@@ -734,9 +748,12 @@ def activate(self):
         # and fix the symlink (in case the VG name changed).
         run.ceph_osd(
             # fmt: off
-            "-i", str(self.id),
-            "--pid-file", self.pid_file,
-            "--osd-data", self.data_volume.mountpoint,
+            "-i",
```
Comment (osnyx) [inline pkgs/fc/ceph/src/fc/ceph/osd/nautilus.py]: formatting

**Review by osnyx — DISMISSED** [review]
LGTM in general, but ruff has again messed up the formatting. Please adjust.

**Review by dpausp — APPROVED** [review]
based on the review by osnyx: the formatting errors have been fixed, so LGTM

**Pull request merged**